contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -265,12 +275,100 @@ abstract contract PoSValidatorManager is
+        if (!_isPoSValidator(validationID)) {
+            revert ValidatorNotPoS(validationID);
+        }
+
+        // Rewards can only be claimed by the validator owner.
+        if ($._posValidatorInfo[validationID].owner != _msgSender()) {
+            revert UnauthorizedOwner(_msgSender());
+        }
+
+        // Check that minimum stake duration has passed.
+        uint64 claimTime = uint64(block.timestamp);
+        if (claimTime < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration) {
+            revert MinStakeDurationNotPassed(claimTime);
+        }
+
+        // The claim's uptime is the difference between the total uptime and the minimum possible uptime from the last claim.
+        // We use the minimum uptime to get a lower bound on the required uptime for this claim
+        uint64 totalUptime = _updateUptime(validationID, messageIndex);
+        uint64 uptime = totalUptime - $._posValidatorInfo[validationID].lastClaimMinUptime;
+
+        // If no rewards have yet been claimed, use the validator's start time
+        uint64 lastClaimTime = $._posValidatorInfo[validationID].lastClaimTime;
+        if (lastClaimTime == 0) {
+            lastClaimTime = validator.startedAt;
+        }
+        // Validate the uptime for this claim. Given that all previous claims have been similarly validated,
+        // this is equivalent to validating the uptime of the entire validation period up to this point, due
+        // to the linearity of the uptime threshold calculation.
+        if (!_validateSufficientUptime(uptime, lastClaimTime, claimTime)) {
+            revert ValidatorIneligibleForRewards(validationID);
+        }
+
+        uint256 reward = $._rewardCalculator.calculateReward({
+            stakeAmount: weightToValue(validator.startingWeight),
+            validatorStartTime: lastClaimTime,
+            stakingStartTime: lastClaimTime,
+            stakingEndTime: claimTime,
+            uptimeSeconds: uptime,
+            initialSupply: 0,
+            endSupply: 0
+        });
+
+        $._posValidatorInfo[validationID].lastClaimMinUptime =
+            (claimTime - validator.startedAt) * UPTIME_REWARDS_THRESHOLD_PERCENTAGE / 100;
+        $._posValidatorInfo[validationID].lastClaimTime = claimTime;
+        _reward($._posValidatorInfo[validationID].owner, reward);
+
+        emit ValidationRewardsClaimed(validationID, reward);
+    }
+
+    function _validateSufficientUptime(
+        uint64 uptimeSeconds,
+        uint64 periodStartTime,
+        uint64 periodEndTime
+    ) internal pure returns (bool) {
+        // Equivalent to uptimeSeconds/(periodEndTime - periodStartTime) < UPTIME_REWARDS_THRESHOLD_PERCENTAGE/100
+        // Rearranged to prevent integer division truncation.
+        if (
+            uptimeSeconds * 100
+                < (periodEndTime - periodStartTime) * UPTIME_REWARDS_THRESHOLD_PERCENTAGE
+        ) {
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
        // Equivalent to uptimeSeconds/(periodEndTime - periodStartTime) >= UPTIME_REWARDS_THRESHOLD_PERCENTAGE/100
        // Rearranged to prevent integer division truncation.
        return uptimeSeconds * 100
                >= (periodEndTime - periodStartTime) * UPTIME_REWARDS_THRESHOLD_PERCENTAGE
```